src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -25,57 +31,100 @@ class AppointmentService(
     durationInMinutes: Int,
     appointmentTime: OffsetDateTime,
     appointmentType: AppointmentType,
-    createdByUser: AuthUser
+    createdByUser: AuthUser,
+    appointmentDeliveryType: AppointmentDeliveryType,
+    appointmentDeliveryAddress: AddressDTO? = null,
   ): Appointment {
 
     val initialAppointmentRequired = appointment == null
     if (initialAppointmentRequired) {
       val deliusAppointmentId = communityAPIBookingService.book(referral, null, appointmentTime, durationInMinutes, appointmentType)
```

Comment:
Double-checking the order of things

1. `communityAPIBookingService.book` with time and context details
2. create appointment and `appointmentRepository.saveAndFlush(appointment)`
3. create appointment delivery and `appointmentRepository.saveAndFlush(appointment)`
4. set the address if applicable and `appointmentDeliveryRepository.saveAndFlush(appointmentDelivery)`

is this ok?

Reply:
Yes, that is the correct order.

Reply:
Yes that is the order in which it occurs.